src/value.cpp

```diff
@@ -29,7 +29,6 @@ void Value::hydrate() {
     case Type::Integer: {
         m_type = Type::Pointer;
         m_object = new IntegerObject { m_integer };
```

Comment:
c++ noob here, but isn't this implying that Value has both m_object and m_integer at the same time? Could it be that when building not in release that is possible and somehow does not complain but when running the release build the union gets compacted and thus you'd end up with some kind of undefined behaviour?

Reply:
It's a good thought. We did try different ordering here, a temporary variable, etc., but it didn't seem to matter. But we may have missed something.